Sources/NostrSDK/Events/GiftWrap/GiftWrapEvent.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  GiftWrapEvent.swift
+//
+//
+//  Created by Terry Yiu on 5/11/24.
+//
+
+import Foundation
+
+/// An event that gift wraps a `SealEvent`.
+/// The wrapped seal is always encrypted to a receiver's pubkey using a random, one-time-use private key.
+/// The gift wrap event tags should include any information needed to route the event to its intended recipient,
+/// including the recipient's `p` tag or [NIP-13 Proof of Work](https://github.com/nostr-protocol/nips/blob/master/13.md).
+///
+/// The underlying `SealEvent` or rumor should not be broadcast by themselves to relays without this gift wrap.
+/// This gift wrap event should be broadcast to only the recipient's relays
+///
+/// See [NIP-59 - Gift Wrap](https://github.com/nostr-protocol/nips/blob/master/59.md).
+public final class GiftWrapEvent: NostrEvent, NIP44v2Encrypting {
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    public init(content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970 - TimeInterval.random(in: 0...172800)), signedBy keypair: Keypair) throws {
+        try super.init(kind: .seal, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    /// Unwraps the content of the gift wrap event and decrypts it into a ``SealEvent``.
+    /// - Parameters:
+    ///   - privateKey: The ``PrivateKey`` to decrypt the content.
+    /// - Returns: The ``SealEvent``.
+    public func unwrap(privateKey: PrivateKey) throws -> SealEvent {
```

Comment:
A function that returns a value should generally be named with a noun phrase describing the result, not a verb phrase describing what it does. This is how most of Apple's APIs are, although `JSONDecoder` is a counterexample. So for this one, it could be something like:
```suggestion
    public func unwrappedSeal(using privateKey: PrivateKey) throws -> SealEvent {
```